.pre-commit-config.yaml

```diff
@@ -36,21 +31,26 @@ repos:
     hooks:
     -   id: pyupgrade
         args: [--py36-plus]
--   repo: https://github.com/asottile/add-trailing-comma
-    rev: v0.8.0
+-   repo: https://gitlab.com/pycqa/flake8
+    rev: 3.7.7
     hooks:
-    -   id: add-trailing-comma
-        args: [--py36-plus]
+    -   id: flake8
+        exclude: ^docs/source/conf.py$
+-   repo: https://github.com/pre-commit/mirrors-autopep8
+    rev: v1.4.3
+    hooks:
+    -   id: autopep8
+-   repo: https://github.com/psf/black
+    rev: stable
```

Comment:
fwiw you shouldn't use floating refs if you want determinism

Reply:
unfortunately, it doesn't seem that `black` has tags for stable releases but only for alpha/beta ones: https://github.com/psf/black/releases

Reply:
right, you want to use one of those, the `stable` tag is just a moving alias for the latest beta